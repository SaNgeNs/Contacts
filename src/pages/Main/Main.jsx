import React, { memo } from 'react';
import Helmet from 'react-helmet/lib/Helmet';
import Form from 'Components/Forms/AddForm';
import CardList from 'Components/CardList';
import { useSelector } from 'react-redux';

export const Main = () => {
  const cards = useSelector(state => state.cards.ids.map(id => {
    const allCards = state.cards.entities;

    return allCards[id];
  }));

  return (
    <div>
      <Helmet>
        <title>Main</title>
        <meta property="og:title" content="Main"/>
        <meta name="twitter:title" content="Main"/>
        <meta name="og:image:alt" content="Main"/>
        <link rel="canonical" href="https://test.com" />
        <meta property="og:url" content="https://test.com" />
        <meta httpEquiv="x-dns-prefetch-control" content="on" />
        <meta name="theme-color" content="#ffffff" />
        <meta property="og:site_name" content="Test" />
        <meta property="og:type" content="page" />
        <meta name="description" content="Main" />
        <meta property="og:description" content="Main" />
      </Helmet>

      <Form />

      <CardList
        cards={cards}
      />
    </div>
  );
};

export default memo(Main);
