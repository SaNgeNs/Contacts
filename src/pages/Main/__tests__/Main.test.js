import React from 'react';
import customRender from 'Mock/customRender';
import MockStore from 'Mock/store';

import Main from '../index';

describe('<Main />', () => {
  it('should render main page', () => {
    const { container, unmount } = customRender(
      <Main />,
      {},
      MockStore,
    );

    expect(container).toMatchSnapshot();
    unmount();
  });
});
