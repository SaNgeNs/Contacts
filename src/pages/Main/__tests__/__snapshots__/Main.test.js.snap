// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should render main page 1`] = `
<div>
  <div />
</div>
`;
